### PR TITLE
declare strict_types in all files

### LINKED ISSUE #464: Declare strict types in all files
This can be done using php-cs-fixer using the [declare_strict_types rule](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/rules/strict/declare_strict_types.rst).

```php
declare(strict_types=1);
```

### PR DESCRIPTION
**Pull request by Art4**
closes #464.

## Summary by CodeRabbit

* **Chores**
  * Enabled strict type checking across the codebase to improve type safety and runtime consistency.
  * Updated legacy listing calls to consistently pass boolean flags.
* **Bug Fixes**
  * Improved XML serialization to consistently treat values and attributes as strings, reducing parsing edge cases.
* **Tests**
  * Test files updated to opt into strict typing to match production behavior.

### CODE REVIEW

File: src/Redmine/Serializer/XmlSerializer.php [hunk 1]
```diff
@@ -75,7 +77,7 @@ private function deserialize(string $encoded): void
 
         try {
             $this->deserialized = new SimpleXMLElement($encoded);
-        } catch (Throwable $e) {
+        } catch (Exception $e) {
```
[the remark this answers is not in the document]
Comment (Art4) [reply src/Redmine/Serializer/XmlSerializer.php:80]: I want to catch only Exceptions thrown by SimpleXMLElement, not TypeError

File: src/Redmine/Api/Search.php [hunk 2]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
```
[the remark this answers is not in the document]
Comment (Art4) [reply src/Redmine/Api/Search.php:3]: I've added the suggested fix by casting $query to string.

**Review by kbsali — APPROVED** [review]
👍
